src/main/java/com/googleintern/wfm/ruleengine/action/CsvParser.java

```diff
@@ -0,0 +1,161 @@
+package src.main.java.com.googleintern.wfm.ruleengine.action;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+import src.main.java.com.googleintern.wfm.ruleengine.model.PoolAssignmentModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.UserPoolAssignmentModel;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * CsvParser class is used to retrieve information from the input csv file ans store these data as
+ * instances of UserPoolAssignmentModel.
+ *
+ * <p>enum variable Header represents the relationship between column position and the data stored
+ * inside. Pattern variable NUMBER_PATTERN finds numbers from strings. Pattern variable
+ * ROLESKILL_PATTERN finds skill_id elements from strings. Pattern variable PERMISSION_PATTERN finds
+ * pool_permission elements from strings.
+ */
+public class CsvParser {
+  enum Header {
+    USER_ID(0),
+    ROLE_ID(1),
+    SKILL_ID(2),
+    ROLESKILL_ID(3),
+    WORKFORCE_ID(4),
+    WORKGROUP_ID(5),
+    POOL_ASSIGNMENT(6);
+
+    final int column;
+
+    Header(int column) {
+      this.column = column;
+    }
+  }
+
+  private static final Pattern NUMBER_PATTERN = Pattern.compile("(\\d+)");
+  private static final Pattern ROLESKILL_PATTERN = Pattern.compile("\"skill_id\":\"(\\d+)\"");
+  private static final Pattern PERMISSION_PATTERN =
+      Pattern.compile("\"cases_pool_id\":\"(\\d+)\",\"permission_set_id\":\"(\\d+)\"");
+
+  /**
+   * Read all data using csvFilePath. Parse useful information out and store it in *
+   * userPoolAssignmentList.
+   *
+   * @param csvFilePath : represents the path to the csv file, including the file name.
+   * @return an ImmutableList of UserPoolAssignmentModel objects
```

Comment:
Javadoc is optional for "simple, obvious" methods. 

https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory